Remove Adobe Flash Player Quirk

Ahmad-S792 opened the pull request:
#### 0f8d06ceb001ea87d985bb546a834014e059267f
<pre>
Remove Adobe Flash Player Quirk

<a href="https://bugs.webkit.org/show_bug.cgi?id=286124">https://bugs.webkit.org/show_bug.cgi?id=286124</a>
<a href="https://rdar.apple.com/143105727">rdar://143105727</a>

Reviewed by Simon Fraser.

This patch removes quirk added back in 2013 (135307@main), where
it was leading to Adobe Flash Player &quot;smaller&quot; installer
firing incorrect load event.

Since Adobe Flash is now end of life, this just removes it.

* Source/WTF/wtf/cocoa/RuntimeApplicationChecksCocoa.h:
* Source/WTF/wtf/cocoa/RuntimeApplicationChecksCocoa.mm:
(WTF::MacApplication::isSolidStateNetworksDownloader): Deleted.
* Source/WebCore/loader/cache/CachedScript.cpp:
(WebCore::CachedScript::shouldIgnoreHTTPStatusCodeErrors const): Deleted.
* Source/WebCore/loader/cache/CachedScript.h:

Canonical link: <a href="https://commits.webkit.org/289110@main">https://commits.webkit.org/289110@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/cdf6f91b24b3b4257b50f37755950f5d41f950bc

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/85366 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/5101 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/39797 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/90494 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/36408 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/87455 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/5191 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/13078 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/66366 "Passed tests") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/24179 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/88412 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/3962 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/77539 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/46648 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/3846 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/64/builds/31800 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/35476 "Built successfully") | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/20/builds/78335 "Built successfully and passed tests") | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/74559 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/32641 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/91992 "Built successfully") | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/12/builds/84407 "Built successfully and passed tests") | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/12713 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/9304 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/74957 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/12942 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/73376 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/74077 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/18332 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/18443 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/119/builds/16874 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/4737 "Built successfully") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/12657 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/18125 "Built successfully") | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/106797 "Built successfully") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/12487 "Built successfully") | | [✅ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/25/builds/25737 "Passed tests") | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/15979 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/14238 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

Ahmad-S792 commented on Source/WebCore/loader/cache/CachedScript.cpp:
```diff
@@ -150,17 +150,6 @@ void CachedScript::setBodyDataFrom(const CachedResource& resource)
 
 bool CachedScript::shouldIgnoreHTTPStatusCodeErrors() const
```
Should we delete this whole function?

karlcow replied:
so let's see. 

https://github.com/WebKit/WebKit/blob/99889a50c5762026c26d6665c88dc28d7c5ee991/Source/WebCore/loader/cache/CachedScript.cpp#L151-L165

this will now return only `CachedResource::shouldIgnoreHTTPStatusCodeErrors()`

So where is `CachedScript::shouldIgnoreHTTPStatusCodeErrors()` used?

https://searchfox.org/wubkat/search?q=shouldIgnoreHTTPStatusCodeErrors&redirect=false

Ahmad-S792 replied:
I think no where (self return), the only thing is that `shouldIgnoreHTTPStatusCodeErrors` in headers (.h) is marked as `final`. Hence, I didn't deleted it, others are `override` or `virtual` functions. If I remove this, it does compile still.

Ahmad-S792 replied:
@brentfulgham - Any thoughts or suggestions?

karlcow replied:
Could It be called sometimes from  https://searchfox.org/wubkat/source/Source/WebCore/loader/SubresourceLoader.cpp

Ahmad-S792 replied:
I think we can safely remove because the definition is here - [CachedResource.h](https://searchfox.org/wubkat/source/Source/WebCore/loader/cache/CachedResource.h) and if we see hierarchy in searchfox.org, we can see it.

I will update this PR and remove this completely.

Ahmad-S792 committed:
Remove Adobe Flash Player Quirk

https://bugs.webkit.org/show_bug.cgi?id=286124
rdar://143105727

Reviewed by Simon Fraser.

This patch removes quirk added back in 2013 (135307@main), where
it was leading to Adobe Flash Player "smaller" installer
firing incorrect load event.

Since Adobe Flash is now end of life, this just removes it.

* Source/WTF/wtf/cocoa/RuntimeApplicationChecksCocoa.h:
* Source/WTF/wtf/cocoa/RuntimeApplicationChecksCocoa.mm:
(WTF::MacApplication::isSolidStateNetworksDownloader): Deleted.
* Source/WebCore/loader/cache/CachedScript.cpp:
(WebCore::CachedScript::shouldIgnoreHTTPStatusCodeErrors const): Deleted.
* Source/WebCore/loader/cache/CachedScript.h:

Canonical link: https://commits.webkit.org/289110@main

darinadler commented on Source/WebCore/loader/cache/CachedScript.h:
```diff
@@ -47,8 +47,6 @@ class CachedScript final : public CachedResource {
 private:
     bool mayTryReplaceEncodedData() const final { return true; }
 
-    bool shouldIgnoreHTTPStatusCodeErrors() const final;
```
Does this function still need to exist in the base class? Does it still need to be a virtual function in the base class?

Ahmad-S792 replied:
I think we can devirtualize it in base class (or try).
